src/open_clip/model.py

```diff
@@ -228,15 +228,35 @@ def encode_image(self, image, normalize: bool = False):
     def encode_text(self, text, normalize: bool = False):
         cast_dtype = self.transformer.get_cast_dtype()
 
-        x = self.token_embedding(text).to(cast_dtype)  # [batch_size, n_ctx, d_model]
+        # text is a list of tensors
+        # get the first dimension of each so we know how to get the right shapes 
+        # after we take the embeddings
+
+        lens = [text[idx].shape[0] for idx in range(len(text))]
```

Comment:
Shouldn't the texts all have the same length as they get padded in the pre-processing?

Reply:
no, only the last dimension is the same